tools/export/cces/__init__.py

```diff
@@ -0,0 +1,473 @@
+        c_flags = self.flags['c_flags'] + self.flags['common_flags']
+        cxx_flags = self.flags['cxx_flags'] + self.flags['common_flags']
+
+        libs = []
+        for libpath in self.resources.libraries:
+            lib = os.path.splitext(os.path.basename(libpath))[0]
+            libs.append(lib[3:]) # skip 'lib' prefix
+
+        ld_flags = self.flags['ld_flags'] + ["-l" + lib for lib \
+                    in self.toolchain.sys_libs]
+
+        proc = self.toolchain.target.device_name
+        cpu = self.toolchain.target.core.lower()
+        fpu = None
+        float_abi = None
+
+        # parse toolchain CPU flags
+        for flag in self.toolchain.cpu:
+            if flag.startswith("-mcpu="):
+                cpu = flag[len("-mcpu="):]
+            elif flag.startswith("-mfpu="):
+                fpu = flag[len("-mfpu="):]
+            elif flag.startswith("-mfloat-abi="):
+                float_abi = flag[len("-mfloat-abi="):]
+
+        # remove toolchain CPU flags. We'll handle them separately
+        # in the generated .json file
+        self.clean_flags(c_flags, self.toolchain.cpu)
+        self.clean_flags(cxx_flags, self.toolchain.cpu)
+        self.clean_flags(ld_flags, self.toolchain.cpu)
+
+        ld_opts = self.convert_linker_options(ld_flags)
+        asm_opts = self.convert_assembler_options(asm_flags)
+        c_opts = self.convert_compiler_options(c_flags)
+        cxx_opts = self.convert_compiler_options(cxx_flags)
+
+        project = "cces"
+        json = "cces.json"
+        local_location = project
+
+        jinja_ctx = {
+            'project' : self.project_name,
+            'cpu' : cpu,
+            'proc' : proc,
+            'family' : "ARM",
+            'asm_defines' : asm_defines,
+            'c_defines' : c_defines,
+            'fpu' : fpu,
+            'float_abi' : float_abi,
+            'ld_script' : ld_script,
+            'local_location' : local_location,
+            'srcs': srcs_dict,
+            'include_dirs' : include_dirs,
+            'ld_opts' : ld_opts,
+            'ld_flags' : ld_flags,
+            'asm_opts' : asm_opts,
+            'asm_flags' : asm_flags,
+            'c_opts' : c_opts,
+            'c_flags' : c_flags,
+            'cxx_opts' : cxx_opts,
+            'cxx_flags' : cxx_flags,
+        }
+
+        self.gen_file('cces/cces.json.tmpl', jinja_ctx,
+                      json, trim_blocks=True, lstrip_blocks=True)
+
+        # generate a readme on how to create the CCES project
+        # using the generated .json file
+
+        jinja_ctx = {
+            'project_create_command' : CCES.get_project_create_command("cces", \
+                                            "WORKSPACE", project),
+            'project_build_command' : CCES.get_project_build_command("cces", \
+                                            "WORKSPACE", project)
+        }
+
+        self.gen_file('cces/README.md.tmpl', jinja_ctx, "README.md")
```

Comment:
As it stands, This file __will__ be generated incorrectly in the online export system. You depend on the **exporting machine's** OS, which will always be Linux in the online compiler, irrespective the OS of the requesting computer. Could you change these files to contain instructions for all 3 OSs?

Reply:
Thanks @theotherjimmy. The resulting README.md should be the same for all OSs - note that the full OS path isn't used for the jinja context since we don't know the client's OS. That said, having instructions for each OS seems reasonable.

Reply:
My mistake. It looks like `get_cces_path` is only called from `build` (which __will__ be called from the same machine invoking the build; no tricks there)